addons/capetrader/capetrader.lua

```diff
@@ -0,0 +1,495 @@
+            if item.id == currentCape.id then
+                capeItem = item
+                break
+            end
+        end
+    end
+
+    local augmentTable
+    if extData.decode(capeItem).augments then
+        augmentTable = extData.decode(capeItem).augments
+    end
+
+    local augValue
+    if augmentTable then
+        if string.contains(string.lower(pathItem.en), 'thread') then
+            augValue = augmentTable[threadIndex]
+        elseif string.contains(string.lower(pathItem.en),'dust') then
+            augValue = augmentTable[dustIndex]
+        elseif string.contains(string.lower(pathItem.en), 'dye') then
+            augValue = augmentTable[dyeIndex]
+        elseif string.contains(string.lower(pathItem.en), 'sap') then
+            augValue = augmentTable[sapIndex]
+        end
+    else
+        augValue = 'none'
+    end
+
+    if string.lower(augValue) == 'none' or not augmentTable then
+        return 'empty'
+    end
+
+    local max = maxAugMap[maxAugKey].max
+    if string.contains(augValue, max) then
+        windower.add_to_chat(123, 'You have augmented your ' .. currentCape.en .. ' to the max already with ' .. pathItem.en .. '.')
+        return 'maxed'
+    end
+
+    local mustContainTable = maxAugMap[maxAugKey].mustcontain
+    for k, augmentString in pairs(mustContainTable) do
+        if not string.contains(string.lower(augValue), string.lower(augmentString)) then
+            windower.add_to_chat(123,'You can\'t augment your ' .. currentCape.en .. ' with ' .. pathName .. ' because it has already been augmented with: ' .. augValue .. ' using ' .. pathItem.en .. '.')
+            return 'notmatching'
+        end
+    end
+
+    local cantContainTable = maxAugMap[maxAugKey].cantcontain
+    if table.length(cantContainTable) > 0 then
+        for k, augmentString in pairs(cantContainTable) do
+            if string.contains(string.lower(augValue), string.lower(augmentString)) then
+                windower.add_to_chat(123,'You can\'t augment your ' .. currentCape.en .. 'with ' .. pathName .. ' because it has already been augmented with: ' .. augValue .. ' using ' .. pathItem.en .. '.')
+                return 'notmatching'
+            end
+        end
+    end
+
+    return 'allclear'
+end
+
+function injectAugmentConfirmationPackets()
+    local optionIndex
+    if firstTimeAug then
+        firstTimeAug = false
+        optionIndex = 512
+    else
+        optionIndex = 256
+    end
+
+    local packet = packets.new('outgoing', 0x05B)
+    packet["Target"] = gorpaID
+    packet["Option Index"] = optionIndex
+    packet["_unknown1"] = pathIndex
+    packet["Target Index"] = gorpaTargetIndex
+    packet["Automated Message"] = true
+    packet["_unknown2"] = 0
+    packet["Zone"] = mhauraID
+    packet["Menu ID"] = gorpaMenuID
+    packets.inject(packet)
+
+    packet["Automated Message"] = false
+    packets.inject(packet)
+
+    packet = packets.new('outgoing', 0x016, {
```

Comment:
I'd recommend choosing a new variable for a new packet. In the above case I'd say it's ok, because the packet is essentially the same, just slightly modified, but here it's entirely new. Having a new variable for it reduces the risk of mixing things up when refactoring the code. Which could be especially problematic for injected outgoing packets.